### PR TITLE
python3Packages.pytest-sanic: 1.6.2 -> 1.7.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 1.7.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 114604` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>python37Packages.entrance</li>
    <li>python37Packages.entrance-with-router-features</li>
    <li>python37Packages.pytest-sanic</li>
    <li>python37Packages.sanic</li>
    <li>python37Packages.sanic-auth</li>
    <li>python38Packages.entrance</li>
    <li>python38Packages.entrance-with-router-features</li>
    <li>python38Packages.pytest-sanic</li>
    <li>python38Packages.sanic</li>
    <li>python38Packages.sanic-auth</li>
  </ul>
</details>

File: pkgs/development/python-modules/pytest-sanic/default.nix [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
   propagatedBuildInputs = [
-    pytest
     aiohttp
     async_generator
+    httpx
+    pytest
+    websockets
   ];
 
   # circular dependency on sanic
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pytest-sanic/default.nix:28]: Maybe we can resolve this circular dependency by using a sanic without tests?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114604` at 20da8e6c run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/62f2d837c0dc1988dcbad40305311eee">python37Packages.entrance</a></li></li>
    <li><a href="https://gist.github.com/599d6f8c0cf3c5fbd2317d2564471d47">python37Packages.entrance-with-router-features</a></li></li>
    <li><a href="https://gist.github.com/03b8f478c12cb81ff167e876c34a3209">python37Packages.pytest-sanic</a></li></li>
    <li><a href="https://gist.github.com/8d66c48bd541512d6797bf2db05a32ad">python37Packages.sanic</a></li></li>
    <li><a href="https://gist.github.com/aae48fe62a417c9880e1fd372e54d7ba">python37Packages.sanic-auth</a></li></li>
    <li><a href="https://gist.github.com/3e2ce35c4b7bd85cad64ccd797f0660a">python38Packages.entrance</a></li></li>
    <li><a href="https://gist.github.com/bf56cf094b60fe7a7cef59360058553e">python38Packages.entrance-with-router-features</a></li></li>
    <li><a href="https://gist.github.com/033fbf16c490bc2b450629af723d265d">python38Packages.pytest-sanic</a></li></li>
    <li><a href="https://gist.github.com/543ee37b9fee3883d53430876ab40894">python38Packages.sanic</a></li></li>
    <li><a href="https://gist.github.com/33b1fdf27da7de00f78b7be9e61fe0de">python38Packages.sanic-auth</a></li></li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114604` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
10 packages built:
</summary>
<ul>
<li>python37Packages.entrance</li>
<li>python37Packages.entrance-with-router-features</li>
<li>python37Packages.pytest-sanic</li>
<li>python37Packages.sanic</li>
<li>python37Packages.sanic-auth</li>
<li>python38Packages.entrance</li>
<li>python38Packages.entrance-with-router-features</li>
<li>python38Packages.pytest-sanic</li>
<li>python38Packages.sanic</li>
<li>python38Packages.sanic-auth</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python37Packages.entrance:
<details>


warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python37Packages.entrance-with-router-features:
<details>


warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.entrance:
<details>


warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.entrance-with-router-features:
<details>


warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
</details>

File: pkgs/development/python-modules/pytest-sanic/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/pytest-sanic/default.nix:28]: Is worth a try.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 114604` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>python38Packages.entrance</li>
    <li>python38Packages.entrance-with-router-features</li>
    <li>python38Packages.pytest-sanic</li>
    <li>python38Packages.sanic</li>
    <li>python38Packages.sanic-auth</li>
  </ul>
</details>

File: pkgs/development/python-modules/pytest-sanic/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/pytest-sanic/default.nix]: I just catched a strange bug with a similar case. This only works if sanic is a test dependency in both packages and not a hard dependency. If it is not the package is not correctly installed and we end up with a mostly empty package in nix.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114604` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>python38Packages.entrance</li>
<li>python38Packages.entrance-with-router-features</li>
<li>python38Packages.pytest-sanic</li>
<li>python38Packages.sanic</li>
<li>python38Packages.sanic-auth</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python38Packages.entrance:
<details>


warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
</details>

File: pkgs/development/python-modules/pytest-sanic/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/pytest-sanic/default.nix]: At first glance it looks like it works but I didn't check the content of the packages.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114604` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>python38Packages.entrance</li>
<li>python38Packages.entrance-with-router-features</li>
<li>python38Packages.pytest-sanic</li>
<li>python38Packages.sanic</li>
<li>python38Packages.sanic-auth</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python38Packages.entrance:
<details>


warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
</details>

**Pull request merged**